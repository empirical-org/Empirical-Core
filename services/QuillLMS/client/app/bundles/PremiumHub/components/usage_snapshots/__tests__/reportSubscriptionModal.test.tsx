import * as React from 'react';
import { render, fireEvent, screen, } from '@testing-library/react';
import ReportSubscriptionModal from '../reportSubscriptionModal';

describe('<ReportSubscriptionModal />', () => {
  const mockSave = jest.fn();
  const mockCancel = jest.fn();
  const existingPdfSubscription = { frequency: 'Weekly' };

  const renderComponent = (isOpen: boolean, subscription = null) =>
    render(
      <ReportSubscriptionModal
        cancel={mockCancel}
        existingPdfSubscription={subscription}
        isOpen={isOpen}
        save={mockSave}
      />
    );

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('matches the component snapshot', () => {
    const { asFragment } = renderComponent(true, existingPdfSubscription);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders when isOpen is true', () => {
    renderComponent(true)
    expect(screen.getByText('Subscribe to this report')).toBeInTheDocument();
  });

  test('does not render when isOpen is false', () => {
    renderComponent(false)
    expect(screen.queryByText('Subscribe to this report')).toBeNull();
  });

  test('initial state with existing subscription', () => {
    renderComponent(true, existingPdfSubscription)
    expect(screen.getByLabelText('On')).toBeChecked();
    expect(screen.getByText('Weekly')).toBeInTheDocument();
  });

  test('sets subscription status to default frequency when clicked', () => {
    renderComponent(true, null)
    fireEvent.click(screen.getByLabelText('On'));
    expect(screen.getByLabelText('On')).toBeChecked();
    expect(screen.getByText('Monthly')).toBeInTheDocument();
  });

  test('calls save function with correct arguments on save click', () => {
    renderComponent(true, existingPdfSubscription)
    fireEvent.click(screen.getByText('Save'));
    expect(mockSave).toHaveBeenCalledWith(true, 'Weekly', existingPdfSubscription);
  });

  test('calls cancel function on cancel click', () => {
    renderComponent(true, existingPdfSubscription)
    fireEvent.click(screen.getByText('Cancel'));
    expect(mockCancel).toHaveBeenCalled();
  });

});

