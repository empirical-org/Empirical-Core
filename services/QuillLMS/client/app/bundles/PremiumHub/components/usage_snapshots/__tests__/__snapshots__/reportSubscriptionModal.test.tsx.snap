// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportSubscriptionModal /> matches the component snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal-container"
  >
    <div
      class="modal-background"
    />
    <div
      class="manage-report-subscription-modal quill-modal modal-body"
    >
      <div>
        <div
          class="title-and-description"
        >
          <h2>
            Subscribe to this report
          </h2>
          <p
            class="description"
          >
            Keep up to date by having this report delivered to your inbox on a recurring basis. It will use filters currently applied when you subscribe, so please review them before doing so.  You can always change your filters later by turning off the report and re-subscribing.
          </p>
        </div>
        <div
          class="is-subscribed-and-frequency"
        >
          <div
            class="is-subscribed-options"
          >
            <h3
              class="is-subscribed-label"
            >
              Email me this report
            </h3>
            <div
              class="radio"
            >
              <label
                id="is-subscribed-on-button"
              >
                <input
                  aria-labelledby="is-subscribed-on-button"
                  type="radio"
                  value="On"
                />
                On
              </label>
            </div>
            <div
              class="radio"
            >
              <label
                id="is-subscribed-off-button"
              >
                <input
                  aria-labelledby="is-subscribed-off-button"
                  checked=""
                  type="radio"
                  value="Off"
                />
                Off
              </label>
            </div>
          </div>
          <div
            class="frequency-options"
          >
            <label
              class="frequency-label"
              for="frequency"
            >
              <span>
                Frequency
              </span>
            </label>
            <div
              class="dropdown-container input-container inactive has-text not-editable disabled  "
              role="button"
              tabindex="0"
            >
              <label />
              <div
                class="dropdown css-b62m3t-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-2-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class="dropdown__control css-13cymwt-control"
                >
                  <div
                    class="dropdown__value-container dropdown__value-container--has-value css-1fdsijx-ValueContainer"
                  >
                    <div
                      class="dropdown__single-value css-1dimb5e-singleValue"
                    >
                      Monthly
                    </div>
                    <input
                      aria-autocomplete="list"
                      aria-expanded="false"
                      aria-haspopup="true"
                      aria-readonly="true"
                      class="css-mohuvp-dummyInput-DummyInput"
                      id="react-select-2-input"
                      inputmode="none"
                      role="combobox"
                      tabindex="-1"
                      value=""
                    />
                  </div>
                  <div
                    class="dropdown__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class="dropdown__indicator-separator css-1u9des2-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class="dropdown__indicator dropdown__dropdown-indicator css-1xc3v61-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="save-and-cancel-buttons"
        >
          <button
            class="quill-button-archived medium secondary outlined focus-on-light"
            type="button"
          >
            Cancel
          </button>
          <button
            class="quill-button-archived medium primary contained focus-on-light"
            type="button"
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
